js/screens/SearchResult/SearchResultScreen.js

```diff
@@ -34,45 +38,136 @@ class SearchResultScreen extends Component {
     };
   };
 
+  getChipLabels() {
+    const {
+      laundryTags,
+      occupantTags,
+      otherTags,
+      parkingTags,
+      propertyTags
+    } = this.props;
+
+    // get all the tags and merge them
+    const allTags = Object.assign(
+      JSON.parse(JSON.stringify(laundryTags)),
+      JSON.parse(JSON.stringify(occupantTags)),
+      JSON.parse(JSON.stringify(otherTags)),
+      JSON.parse(JSON.stringify(parkingTags)),
+      JSON.parse(JSON.stringify(propertyTags))
+    );
+
+    const trueTags = getTrueParams(allTags);
+    return trueTags;
+  }
+
+  closeChip(tagName) {
+    const {
+      dispatch,
+      laundryTags,
+      occupantTags,
+      otherTags,
+      parkingTags,
+      propertyTags
+    } = this.props;
+
+    const tagSet = setParamToFalse(tagName, [
+      { updateLaundryTags: laundryTags },
+      { updateOccupantTags: occupantTags },
+      { updateOtherTags: otherTags },
+      { updateParkingTags: parkingTags },
+      { updatePropertyTags: propertyTags }
+    ]);
+
+    // set the tagSet to false in redux
+    const functionName = Object.keys(tagSet)[0];
+    const tags = Object.values(tagSet)[0];
+
+    switch (functionName) {
+      case 'updateLaundryTags':
+        dispatch(updateLaundryTags({ ...tags }));
+        break;
+      case 'updateOccupantTags':
+        dispatch(updateOccupantTags({ ...tags }));
+        break;
+      case 'updateOtherTags':
+        dispatch(updateOtherTags({ ...tags }));
+        break;
+      case 'updateParkingTags':
+        dispatch(updateParkingTags({ ...tags }));
+        break;
+      case 'updatePropertyTags':
+        dispatch(updatePropertyTags({ ...tags }));
+        break;
+    }
+    queryBasedOnFilters();
```

Comment:
if the query is based on filters why not pass filter tags with this function.

Reply:
because the tags are stored in redux and I don't want to pass 6 things every time I use this function.

Reply:
you can use default arguments and pass the ones you require.